Code/Tools/ProjectManager/Source/FormLineEditTagsWidget.cpp

```diff
@@ -0,0 +1,201 @@
+        tagsLayout->addStretch();
+        
+        m_tagFrame->setLayout(tagsLayout);
+        m_tagFrame->setVisible(false);
+        tagsLayout->setSizeConstraint(QLayout::SetNoConstraint);
+
+        m_mainLayout->addWidget(m_tagFrame);
+    }
+
+    FormLineEditTagsWidget::FormLineEditTagsWidget(const QString& labelText, const QString& valueText, QWidget* parent)
+        : FormLineEditTagsWidget(labelText, valueText, "", "", parent)
+    {
+    }
+
+    void FormLineEditTagsWidget::forceSetText(const QString& text)
+    {
+        m_lineEdit->setText(text);
+        m_lineEdit->setFocus();
+    }
+
+    void FormLineEditTagsWidget::forceSubmitCurrentText()
+    {
+        addToTagList(m_lineEdit->text());
+        m_lineEdit->clear();
+        m_lineEdit->setText("");
+        refreshTagFrame();
+    }
+
+    void FormLineEditTagsWidget::refreshTagFrame()
+    {
+        // cleanup the tag frame widget and re-add the tag list
+        qDeleteAll(m_tagFrame->children());
+        QHBoxLayout* layout = new QHBoxLayout(this);
+        layout->setSpacing(8);
+
+        for (auto t : m_tags)
+        {
+            QCheckBox* tc = new QCheckBox(t, this);
+            tc->setLayoutDirection(Qt::RightToLeft);
+            // connect the checked signal to a good slot
+            connect(tc, &QCheckBox::stateChanged, this, &FormLineEditTagsWidget::processTagDelete);
+            layout->addWidget(tc);
+        }
+
+        layout->addStretch();
+        layout->setSizeConstraint(QLayout::SetNoConstraint);
+        m_tagFrame->setLayout(layout);
+
+        bool makeVisible = m_tags.count() > 0;
+        m_tagFrame->setVisible(makeVisible);
+        refreshStyle();
+    }
+
+    void FormLineEditTagsWidget::addToTagList(const QString& text)
+    {
+        if((QString::compare(text,"") > 0 || QString::compare(text,"") < 0) && !m_tags.contains(text))
+        {
+            m_tags << text;
+        }
+    }
+
+    void FormLineEditTagsWidget::textChanged(const QString& text)
+    {
+        if(text.contains(" "))
+        {
+            //do tag adding
+            QString interim = m_lineEdit->text();
+            m_lineEdit->clear();
+            for(auto str : interim.split(" "))
+            {
+                addToTagList(str);
+            }
+            refreshTagFrame();
+        }
+    } 
+
+    void FormLineEditTagsWidget::processTagDelete([[maybe_unused]] int unused)
+    {
+        auto checkBoxes = m_tagFrame->findChildren<QCheckBox* >();
+        QString string;
+        bool found = false;
+        for(auto c : checkBoxes)
+        {
+            if(c->isChecked())
+            {
+                string = c->text();
+                found = true;
+                break;
+            }
+        }
+
+        if(!found)
+        {
+            return;
+        }
+        //remove the offending string
+        m_tags.removeOne(string);
+
+        //now reconstruct the tag list
+        refreshTagFrame();
+    }
```

Comment:
Since the individual QCheckbox that was clicked is sending the signal, you should be able to just use `sender()` to get a pointer to it and avoid the loop to search?